azurecompute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeServiceAdapter.java

```diff
@@ -88,17 +339,70 @@ public void resumeNode(String id) {
    @Override
    public void suspendNode(String id) {
       // TODO Auto-generated method stub
-
    }
 
    @Override
    public Iterable<Deployment> listNodes() {
-      // TODO Auto-generated method stub
-      return null;
+      Set<Deployment> deployments = FluentIterable.from(api.getCloudServiceApi().list())
+              .transform(new Function<CloudService, Deployment>() {
+                 @Override
+                 public Deployment apply(CloudService cloudService) {
+                    return api.getDeploymentApiForService(cloudService.name()).get(cloudService.name());
+                 }
+              })
+              .filter(notNull())
+              .toSet();
+      return deployments;
    }
 
    @Override public Iterable<Deployment> listNodesByIds(Iterable<String> ids) {
       // TODO Auto-generated method stub
       return null;
```

Comment:
If there is no method in the provider API to get N nodes by their ID, just call the `listNodes()` and filter the results.